components/search_hint/search_hint.tsx

```diff
@@ -1,32 +1,84 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {PureComponent} from 'react';
-import {FormattedMessage} from 'react-intl';
+import React, {useState, useEffect} from 'react';
+import {FormattedMessage, MessageDescriptor} from 'react-intl';
+import classNames from 'classnames';
 
-import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
+import {t} from 'utils/i18n';
+
+interface SearchTerm {
+    searchTerm: string;
+    message: MessageDescriptor;
+}
 
 type Props = {
     withTitle?: boolean;
+    updateSearchTerms: (term: string) => void;
+    onMouseDown?: () => void;
+    options: SearchTerm[];
 }
 
-export default class SearchHint extends PureComponent<Props> {
-    public render() {
-        return (
-            <React.Fragment>
-                {this.props.withTitle &&
-                <h4>
+export const SearchHint = (props: Props) => {
```

Comment:
Thanks for changing that. I could've sworn you could do `export default const Foo = ...` all on one line, but I guess not. At the very least this component can be imported like it would in other places now.